File: client/src/app/_models/actionItemParams.ts
import { ActionItem } from "./actionItem";
import { User } from "./user";

export class ActionItemParams {
    pageNumber = 1;
    pageSize = 5;
    mapStatus = "all";
    orderBy = "created";
    actionItemNumber = "";
    taskNumber = "";
    internalEmailSubject = "";
    externalEmailSubject = "";
    workOrderTypeRequest = "";
    metSLO = "";
    metElapsedTarget = "";
    dateFromStarted = "";
    dateToStarted = ""
    dateFromResolved = "";
    dateToResolved = "";

    constructor(user?: User) {
        //this.gender = user.gender === 'female' ? 'male' : 'female';
    }
}

File: API/Helpers/ActionItemParams.cs
using System;

namespace API.Helpers
{
    public class ActionItemParams : PaginationParams
    {
        public string CurrentUsername { get; set; }
        public string InternalEmailSubject { get; set; }
        public string ExternalEmailSubject { get; set; }
        public string TaskNumber { get; set; }
        public string ActionItemNumber { get; set; }
        public string OrderBy { get; set; } = "created";
        public string WorkOrderTypeRequest { get; set; }
        public string MapStatus { get; set; } = "all";
        public int AICreatedBy { get; set; }
        public string MetSLO { get; set; }
        public string MetElapsedTarget { get; set; }
        public string DateStartedFrom { get; set; }
        public string DateStartedTo { get; set; }
        public string DateResolvedFrom { get; set; }
        public string DateResolvedTo { get; set; }
    }
}